src/main/java/io/kestra/plugin/fs/ssh/Command.java

```diff
@@ -46,22 +45,43 @@
 @Plugin(
     examples = {
         @Example(
+            title = "Run SSH command using password authentication",
             code = {
                 "host: localhost",
                 "port: 22",
+                "authMethod: PASSWORD",
                 "username: foo",
                 "password: pass",
-                "command: ls",
+                "commands: ['ls']",
+            }
+        ),
+        @Example(
+            title = "Run SSH command using public key authentication",
+            code = {
+                "host: localhost",
+                "port: 22",
+                "authMethod: PUBLIC_KEY",
+                "username: root",
+                "privateKey: secret('SSH_RSA_PRIVATE_KEY')",
+                "commands: ['touch kestra_was_here']"
             }
         )
     }
 )
-public class Command extends Task implements AbstractVfsInterface, RunnableTask<Command.ScriptOutput> {
+public class Command extends Task implements SshInterface, RunnableTask<Command.ScriptOutput> {
     private static final long SLEEP_DELAY_MS = 25L;
 
     private String host;
+    private AuthMethod authMethod;
+
+    // Password Auth method
     private String username;
-    private String password;
+    private Optional<String> password;
+
+    // PubKey Auth method
+    private Optional<String> privateKey;
+    private Optional<String> privateKeyPassphrase;
```

Comment:
We usually don't use Optional but check for not null instead.
I agree that Optional is convenient but please remove them for consistency with other plugins.